sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/query/CosmosQuery.java

```diff
@@ -72,15 +74,37 @@ public int getLimit() {
         return limit;
     }
 
+    /**
+     * To get offset number
+     *
+     * @return long offset
+     */
+    public long getOffset() {
+        return offset;
+    }
+
     /**
      * To set limit number
      *
      * @param limit int
      */
-    public void setLimit(int limit) {
+    public void withLimit(int limit) {
```

Comment:
We cannot change the existing API. The correct way to do this is to deprecate the existing API, (keep it - with an annotation), and add the new API.

Reply:
Done :)